alchemy.lic

```diff
@@ -0,0 +1,164 @@
+    elsif args.craft
+      wait_for_script_to_complete('remedy', ['remedies', recipe['chapter'], recipe['name'], recipe['herb1'], recipe['herb2'], settings['catalyst'], recipe['container'], recipe['noun']])
+    end
+  end
+
+
+  def forage_ingredient(ingredient)
+    #check for sufficient ranks
+    echo "insufficient ranks to forage #{ingredient['name']}" if DRSkill.getrank('Outdoorsmanship') < ingredient['ranks']
+    exit if DRSkill.getrank('Outdoorsmanship') < ingredient['ranks']
+    #check for seasons
+    #check for time of day
+    DRCA.check_moonwatch
+    echo UserVars.sun['day'] if @debug
+    echo ingredient['times'].find { |key| key = 'day' } if @debug
+#    exit unless UserVars.sun['night'] and ingredient['times'].find { |key| key = 'night' }
+#    exit if UserVars.sun['day'] and ingredient['times'].find { |key| key = 'day' }
+#    DRCA.check_moonwatch
+#    moondata = UserVars.moons.find { |moon_name, moon_data| UserVars.moons['visible'].include?(moon_name) && moon_data['timer'] >= 4 }.first
+    echo ("herb_quantity_needed = #{@herb_quantity_needed}") if @debug
+    echo ("alchemy_herb_quantity = #{@alchemy_herb_quantity}") if @debug
+    if @herb_quantity_needed > 0
+      t_quantity = @herb_quantity_needed
+    else
+      t_quantity = @alchemy_herb_quantity
+    end
+    echo ("t_quantity = #{t_quantity}") if @debug
+    # Set this variable to help with counting past 120 - 20 * 6.
+    quantity_needed = (t_quantity - count(ingredient['name'],@alchemy_herb_storage))
+    echo quantity_needed if @debug
+    return if quantity_needed <= 0
+    stow_hands
+    walk_to(@location)
+    while quantity_needed >= 0
+      snapshot = "#{right_hand}#{left_hand}"
+      while snapshot == "#{right_hand}#{left_hand}"
+        case bput("forage #{ingredient['name']} #{@alchemy_forage_type}", 'Roundtime', 'The room is too cluttered to find anything here', 'You really need to have at least one hand free to forage properly', 'You survey the area and realize that any foraging efforts would be futile')
+        when 'The room is too cluttered to find anything here'
+          return false unless kick_pile?
+        when 'You survey the area and realize that any foraging efforts would be futile'
+          return false
+        when 'You really need to have at least one hand free to forage properly'
+          echo 'WARNING: hands not emptied properly. Stowing...'
+          fput('stow right')
+        end
+        waitrt?
+      end
+      quantity_needed -= 6
+      fput("put my #{ingredient['name']} in my #{@alchemy_herb_storage}")
+      echo quantity_needed if @debug
+    end
+  end
+
+  def prepare_ingredient(ingredient)
+    #dry? crush?
+    stow_hands
+    exit if @alchemy_prep_quantity <= count(ingredient['output'],@herb_container)
+    find_press_grinder_room(@press_location)
```

Comment:
💡 `DRCC.` please

Reply:
I have common-crafting and include DRCC. Do I still need to prefix it with `DRCC.`?

Reply:
Need to, no. But it makes it clear where functions are coming from, i.e. easier for new people to get the hang of what's going on.